src/Microsoft.DotNet.ProjectJsonMigration/Rules/MigrateTFMRule.cs

```diff
@@ -0,0 +1,93 @@
+﻿using System;
+using System.Text;
+using System.Globalization;
+using Microsoft.Build.Construction;
+using System.Linq;
+using NuGet.Frameworks;
+
+namespace Microsoft.DotNet.ProjectJsonMigration
+{
+    // TODO: Support Multi-TFM
+    public class MigrateTFMRule : IMigrationRule
+    {
+        private readonly ITransformApplicator _transformApplicator;
+        private readonly AddPropertyTransform<NuGetFramework>[] _transforms;
+
+        public MigrateTFMRule(TransformApplicator transformApplicator = null)
+        {
+            _transformApplicator = transformApplicator ?? new TransformApplicator();
+
+            _transforms = new AddPropertyTransform<NuGetFramework>[]
+            {
+                OutputPathTransform,
+                FrameworkIdentifierTransform,
+                FrameworkVersionTransform
+            };
+        }
+
+        public void Apply(MigrationSettings migrationSettings, MigrationRuleInputs migrationRuleInputs)
+        {
+            var csproj = migrationRuleInputs.OutputMSBuildProject;
+            var propertyGroup = migrationRuleInputs.CommonPropertyGroup;
+
+            CleanExistingProperties(csproj);
+
+            foreach (var transform in _transforms)
+            {
+                _transformApplicator.Execute(
+                    transform.Transform(migrationRuleInputs.DefaultProjectContext.TargetFramework),
+                    propertyGroup);
+            }
+        }
+
+        private void CleanExistingProperties(ProjectRootElement csproj)
+        {
+            var existingPropertiesToRemove = new string[] { "TargetFrameworkIdentifier", "TargetFrameworkVersion" };
+            foreach (var propertyName in existingPropertiesToRemove)
```

Comment:
Why the double loop here? Why not update the where clause below with something like existingPropertiesToRemove.Contains(p.Name)?